### PR TITLE
feature/1077 - Update flight strips view with Arrivals and Departures lists

### LINKED ISSUE #1077: Update flight strips view by separating arrival and departure strips
Currently, all arrival and departure strips are combined in the same list. It would be nice to seperate them into Arrivals and Departures lists for easier management.

If this feature is agreed to be added, I currently have it working in a private branch. If I can be added to this repo org, I can push it to a branch here.

Related: #1088.

### PR DESCRIPTION
**Pull request by tdawe**
### **[Test it here](https://openscope-dev-pr-1078.herokuapp.com/)**

Resolves #1077.
Related: #1088.

The purpose of this pull request is to update the flight strips view by separating Arrivals and Departures into their own lists.

arrivals-departures

### CODE REVIEW

**Comment by felixscheffer** [comment]
looks good to me

**Comment by indianbhaji** [comment]
Needs a changelog

**Comment by Marcel510** [comment]
Just tried it out works very good and really improves gameplay, thank you for your work.
Make sure to request a review as soon as it's ready for review 😄

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js [hunk 1]
```diff
@@ -149,7 +158,7 @@ export default class StripViewController {
             let stripViewModel = this._collection.findStripByAircraftId(aircraftModel.id);
             
             if (typeof stripViewModel === "undefined") {
-                stripViewModel = this.createStripView(aircraftModel);
+                stripViewModel = this.createStripView(aircraftModel, aircraftModel.isDeparture());
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewController.js]: the `aircraft.isDeparture()` method is unnecessary here.  since `.createStripView()` is receiving the entire `AircraftModel`, you don't need to pass it separately.  the receiving method call it directly.

File: src/assets/style/landmark/_stripView.less [hunk 2]
```diff
@@ -18,18 +18,19 @@
     right: -390px;
 }
 
+.stripTitle {
```
Comment (n8rzz) [inline src/assets/style/landmark/_stripView.less]: I'd like to see a little extra padding around the title.  Try something like this:
```css
.stripTitle {
    background-color: #222222;
    color: #ffffff;
    font-size: 1.25em;
    padding: 5px 0;
    text-align: center;
}
```

I'd also like to see existing color vars used here instead of inline hex codes.

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js [hunk 3]
```diff
@@ -162,11 +171,12 @@ export default class StripViewController {
      * @for StripViewController
      * @method createStripView
      * @param aircraftModel {AircraftModel}
+     * @param isDeparture {boolean}
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewController.js]: remove

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js [hunk 4]
```diff
@@ -162,11 +171,12 @@ export default class StripViewController {
...
      * @return {StripViewModel}
      */
-    createStripView(aircraftModel) {
+    createStripView(aircraftModel, isDeparture) {
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewController.js]: remove `isDeparture` param

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js [hunk 5]
```diff
@@ -162,11 +171,12 @@ export default class StripViewController {
...
         const stripViewCid = this._generateCidNumber();
-        const stripViewModel = new StripViewModel(aircraftModel, stripViewCid);
+        const stripViewModel = new StripViewModel(aircraftModel, stripViewCid, isDeparture);
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewController.js]: remove `isDeparture` arg

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js [hunk 6]
```diff
@@ -273,11 +283,11 @@ export default class StripViewController {
             throw new TypeError(`Expected an instance of StripViewModel but received ${typeof stripViewModel}`);
         }
 
-        const scrollPosition = this.$stripViewList.scrollTop();
-
-        this.$stripViewList.append(stripViewModel.$element);
+        const listView = stripViewModel.isDeparture ? this.$stripViewListDepartures : this.$stripViewListArrivals
+        const scrollPosition = listView.scrollTop();
+        listView.append(stripViewModel.$element);
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewController.js]: missing empty line between 287/288

File: src/assets/scripts/client/aircraft/StripView/StripViewModel.js [hunk 7]
```diff
@@ -48,8 +48,9 @@ export default class StripViewModel extends BaseModel {
      * @constructor
      * @param aircraftModel {object}
      * @param cidValue {number}
+     * @param isDeparture {boolean}
      */
-    constructor(aircraftModel, cidValue) {
+    constructor(aircraftModel, cidValue, isDeparture) {
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: remove `isDeparture` param

File: src/assets/scripts/client/aircraft/StripView/StripViewModel.js [hunk 8]
```diff
@@ -69,6 +70,17 @@ export default class StripViewModel extends BaseModel {
...
+        /**
+         * If an aircraft is a Departure
+         *
+         * When this is true it means the aircraft is a Departure
+         * and if this is false it means the aircraft is an Arrival.
+         *
+         * @property isDeparture
+         * @type {boolean}
+         */
+        this.isDeparture = isDeparture;
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: default this to `false`

Comment (n8rzz) [reply src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: also, since the changeset doesn't extend down to the `._init()` method, add the assignment of `#isDeparture` in that method by using the passed `aircraftModel` reference.

File: src/assets/style/landmark/_stripView.less [hunk 9]
```diff
@@ -18,18 +18,19 @@
...
+    color: #ffffff;
+    font-size: 2em;
+    background-color: #222222;
```
Comment (n8rzz) [inline src/assets/style/landmark/_stripView.less]: be sure to alphabetize css properties

File: src/assets/style/landmark/_stripView.less [hunk 10]
```diff
@@ -18,18 +18,19 @@
...
 .stripView-list {
-    max-height: 100%;
     // LESS itself tries to evaluate calc(), using ~"" here to prevent that
     width: ~"calc(100% - " @spacing-2x ~")";
-    position: absolute;
-    top: @spacing-2x;
-    bottom: @spacing-2x;
-    left: @spacing-2x;
-    display: flex;
-    flex-direction: column-reverse;
```
Comment (n8rzz) [inline src/assets/style/landmark/_stripView.less]: Why did you remove the `column-reverse`?  The idea here was to have the list populate from the bottom up, just like a real flight strip list.  By removing that, each list will populate from the top down.

@erikquinn @Fechulo are you guys of with that change?  It's a slight departure from realism, though probably makes things a little easier to manage code-wise.

File: src/templates/_stripView.hbs [hunk 11]
```diff
@@ -1,4 +1,7 @@
 <div id="sidebar" class="stripView js-stripView mix-stripView_isHidden">
     <div class="stripView-trigger js-stripView-trigger"></div>
-    <ul id="strips" class='stripView-list nice-scrollbar js-stripView-list'></ul>
+    <div class="stripTitle">Arrivals</div>
+    <ul id="strips-arrivals" class='stripView-list nice-scrollbar js-stripViewArrivals-list'></ul>
```
Comment (n8rzz) [inline src/templates/_stripView.hbs]: are you using the `id`s at all?  if not, remove them.  if you are, define them as classes instead because we are slowly moving away from the `id`s.

File: src/assets/style/landmark/_stripView.less (hunk 2 above)
Comment (tdawe) [reply src/assets/style/landmark/_stripView.less]: Fixed in latest commit

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js (hunk 4 above)
Comment (tdawe) [reply src/assets/scripts/client/aircraft/StripView/StripViewController.js]: Fixed in latest commit

File: src/assets/scripts/client/aircraft/StripView/StripViewModel.js (hunk 7 above)
Comment (tdawe) [reply src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: Fixed in latest commit

File: src/assets/scripts/client/aircraft/StripView/StripViewModel.js (hunk 8 above)
Comment (tdawe) [reply src/assets/scripts/client/aircraft/StripView/StripViewModel.js]: Fixed in latest commit

File: src/assets/style/landmark/_stripView.less (hunk 9 above)
Comment (tdawe) [reply src/assets/style/landmark/_stripView.less]: Fixed in latest commit

File: src/assets/style/landmark/_stripView.less (hunk 10 above)
Comment (tdawe) [reply src/assets/style/landmark/_stripView.less]: I reverted this back to ``column-reverse``

File: src/templates/_stripView.hbs (hunk 11 above)
Comment (tdawe) [reply src/templates/_stripView.hbs]: No, not using these ids. Removed them from latest commit.

File: src/assets/scripts/client/aircraft/StripView/StripViewController.js (hunk 6 above)
Comment (tdawe) [reply src/assets/scripts/client/aircraft/StripView/StripViewController.js]: Added newline

**Comment by tdawe** [comment]
@n8rzz, all issues have been resolved. Please let me know if you have any other required changes for this pull request.

@erikquinn, let me know if you have any other required changes as well. Thanks!

**Comment by erikquinn** [comment]
Nice job on this! Added #1088 as a future improvement to be able to slide the divider up/down so we can have more of a choice over how our screen space is to be utilized.

Thanks for the work on this one!
